src/NHibernate/Cache/ReadOnlyCache.cs

```diff
@@ -11,34 +11,39 @@ namespace NHibernate.Cache
 	/// </summary>
 	public partial class ReadOnlyCache : IBatchableCacheConcurrencyStrategy
 	{
+#pragma warning disable 618
 		private ICache cache;
-		private IBatchableReadOnlyCache _batchableReadOnlyCache;
-		private IBatchableCache _batchableCache;
+#pragma warning restore 618
+		private CacheBase _batchableCache;
 		private static readonly INHibernateLogger log = NHibernateLogger.For(typeof(ReadOnlyCache));
 
 		/// <summary>
 		/// Gets the cache region name.
 		/// </summary>
 		public string RegionName
 		{
-			get { return cache.RegionName; }
+			get { return Cache.RegionName; }
 		}
 
+		// 6.0 TODO: type as CacheBase instead and switch to autoproperty
+#pragma warning disable 618
 		public ICache Cache
```

Comment:
I thin this needs to be made explicit.

Reply:
It would be a binary breaking change. Setting an `ReadOnlyCache.Cache` with an `ICache` not deriving from `CacheBase` would no more compile.

Reply:
So, are we talking about a binary breaking change (drop-in replacement) or a source breaking change (when it does not compile anymore)? I don't think it will create a binary breaking change.

Reply:
It is a binary breaking change for anyone using the concrete type, which is public. Of course no one is supposed to use that concrete type directly. But for ensuring this it should have been internal.
Unfortunately it is not, so that is still a binary breaking change. It will go unnoticed by most users, but maybe not by some.